src/main/java/mekanism/common/base/IFluidHandlerWrapper.java

```diff
@@ -1,21 +1,46 @@
 package mekanism.common.base;
 
+import javax.annotation.Nonnull;
 import javax.annotation.Nullable;
 import net.minecraft.util.EnumFacing;
 import net.minecraftforge.fluids.FluidStack;
 import net.minecraftforge.fluids.FluidTankInfo;
 
 public interface IFluidHandlerWrapper {
 
-    int fill(EnumFacing from, @Nullable FluidStack resource, boolean doFill);
-
-    FluidStack drain(EnumFacing from, @Nullable FluidStack resource, boolean doDrain);
-
-    FluidStack drain(EnumFacing from, int maxDrain, boolean doDrain);
-
-    boolean canFill(EnumFacing from, @Nullable FluidStack fluid);
-
-    boolean canDrain(EnumFacing from, @Nullable FluidStack fluid);
+    /**
+     * It is assumed that canFill is checked before calling this method
+     */
+    default int fill(EnumFacing from, @Nonnull FluidStack resource, boolean doFill) {
+        return 0;
+    }
+
+    /**
+     * It is assumed that canDrain is checked before calling this method
+     */
+    @Nullable
+    default FluidStack drain(EnumFacing from, @Nonnull FluidStack resource, boolean doDrain) {
+        return drain(from, resource.amount, doDrain);
+    }
+
+    /**
+     * It is assumed that canDrain is checked before calling this method
+     */
+    @Nullable
+    default FluidStack drain(EnumFacing from, int maxDrain, boolean doDrain) {
+        return null;
+    }
+
+    default boolean canFill(EnumFacing from, @Nonnull FluidStack fluid) {
+        return false;
+    }
+
+    /**
+     * null fluid needs to have amounts end up getting specified for draining rather than using the drain with a null stack
```

Comment:
I don't understand this comment

Reply:
Actually it sort of is invalid/doesn't make sense after I changed some nullability declarations for other methods.